TS3 update (#351)
eu-digital-identity-wallet/eudi-doc-standards-and-technical-specifications: docs/technical-specifications/ts3-wallet-unit-attestation.md

| Life Cycle          | Issued by Wallet Providers that must maintain revocation status for the validity period of the WUA.                                                                                                                             |
| Revocation          | Revocation Status List.                                                                                                                                                                                                        |

Below we present the current technical specification for Issuance WUAs in more depth as well as discuss some alternative choices. 

## 2.1 Format
A Wallet Unit Attestation SHALL be a JSON Web Token (JWT) as specified in [RFC7519](https://datatracker.ietf.org/doc/html/rfc7519).

## 2.2 Transport
The ARF specifies that OID4VCI is to be used to transport the WUA used for issuance, hence the format of the WUA shall be supported by OID4VCI. 

### 2.2.1 Options within the OID4VCI Protocol

> Before, discussing the options for transport of Issuance WUAs, we briefly sketch the OID4VCI protocol. There are three entities in the protocol: 

Reviewer: The list below contains 4 items/entities ?

> Before, discussing the options for transport of Issuance WUAs, we briefly sketch the OID4VCI protocol. There are three entities in the protocol: 

Author: ```suggestion
Before, discussing the options for transport of Issuance WUAs, we briefly sketch the OID4VCI protocol. There are four entities in the protocol: 
```

3. *Wallet* sends a  *Credential Request* to the *Credential Issuer* along with the previously obtained access token.
4. *Credential Issuer* sends a *Credential Response* (containing the issued credential) to the *Wallet*. 

Further details can be found here [OIDF OpenID for Verifiable Credential Issuance - draft 15](https://github.com/eu-digital-identity-wallet/eudi-doc-standards-and-technical-specifications/issues/3).

The OID4VCI protocol allows to transfer attestations from Wallet Providers to Issuers (Attestation Providers) through Wallets at two different levels: 
1. They can be sent along the Authorization Request and the Token Request from the Wallet to the Authorization Server (i.e., in step 2 of the above protocol sketch) as a [OAuth 2.0 Attestation-Based Client Authentication](https://datatracker.ietf.org/doc/draft-ietf-oauth-attestation-based-client-auth/). This is referred to as a Wallet Attestation. It is worth noting that that OAuth 2.0 Attestation-Based Client Authentication must include a Proof-of-Possession to a *Client Instance Key* key also appearing in the Wallet Attestation.

> 2. They can be sent in the Credential Request in a `proofs` field from Wallet to Issuance Server (i.e., in step 4 of the above protocol sketch).

Author: ```suggestion
2. They can be sent in the Credential Request in a `proofs` field from Wallet to Credential Issuer (i.e., in step 4 of the above protocol sketch).
```

| Short lived attestations | As an alternative to revocation, the WUA can expire less than 24 hours after issuance, this would eliminate the need for revocation altogether. | WP does not need to maintain revocation status. | WP will often have to perform verifications of the integrity of the Wallet Unit. Short lived attestations will not accomodate the legal requirement for PID providers to check the revocation status of the Wallet Unit. |

### 2.5.2 Revocation proposal

With the above solutions in mind, we propose to use the chunked status list described in Section 2.5.1, i.e. a separate status list is generated for all WUAs issued each single day. The URI will have a format similar to: 

> `https://revocation_url/statuslists/2024/12/24/` for WUAs issued on Dec. 24, 2024. We note that status lists encourage compression, which will reduce the size in general. It is also possible to do additional chunking to further reduce the status list sizes. A similar approach to revocation status list are used elsewhere in the ARF and it would prefered if the same method could also be used for WUA.

Reviewer: ```suggestion
`https://revocation_url/statuslists/2024/12/24/` for WUAs issued on Dec. 24, 2024. We note that status lists encourage compression, which will reduce the size in general. It is also possible to do additional chunking to further reduce the status list sizes. A similar approach to revocation status list are used elsewhere in the ARF and it would be preferable if the same method could also be used for WUA.
```
